### ISSUE TITLE
Some carousels not displaying correctly

### ISSUE THREAD

**Issue by MignonBelongie** [internetarchive/openlibrary#5915]
Some carousels don't have scroll arrows and are displaying narrower images than they should be.

### Evidence / Screenshot (if possible)
![retro-bug](https://user-images.githubusercontent.com/1554660/143922886-86a38ab7-9749-410b-9b71-c2a39cd7ce21.png)

### Relevant url?
https://openlibrary.org/collections/hugo-awards

### Steps to Reproduce

1. Go to above URL and scroll to the bottom.
2. Look at the two Retro-Hugo carousels (which are both based on the same query with eight results, one newest first and one oldest first, just to make sure all works are visible in one or the other).

* Actual:
* Expected:

### Details

- **Logged in (Y/N)?**
- yes
- **Browser type/version?**
- Brave
- **Operating system?**
- windows
- **Environment (prod/dev/local)?** 
- prod

**Comment by MignonBelongie** [comment]
@cdrini I don't see that error in my js console.

I edited the collection a little (it now has three retro-hugo carousels, all broken, followed by a non-retro-hugo carousel, which displays correctly) just to rule out the possibility that a single item is somehow breaking the carousels.

**Comment by MignonBelongie** [comment]
This bug turns out to be due to having two carousels with the same key, which can happen either due to specifying the same key in two carousels, or, as in my case, not specifying keys, and having two carousels with the same query (since keys are generated from a hash of the query when no key is specified). Note that the problem doesn't affect the "duplicate" carousels, but rather breaks any subsequent carousels (more precisely, any subsequent carousels that are not themselves a "duplicate" of a previous carousel).

This could be fixed by having some kind of registry for carousels, so that a unique key can always be generated, but that's probably overkill. The workaround, of course, is just to specify keys explicitly if you want to have multiple carousels with the same query. I don't have permission to edit https://openlibrary.org/help/faq/collections-carousel-examples, so I suggest that someone add the following to the section called "key=?":

"Since keys must be unique, if you have multiple carousels with the same query, you must assign keys for them."

To address my previous comment, where I had modifed the collection so that I had three different retro-hugo carousels, all broken, followed by a non-retro-hugo carousel, which displayed correctly: The retro-hugo carousels were all failing only because they came after two carousels with the same query. The carousel I added after them was a copy of a previous carousel, i.e. a "duplicate" carousel, and thus was correctly displayed (because it was generated at the same time as the original).
